examples/visionProcessing/faceDetection/faceDetection.cpp

```diff
@@ -0,0 +1,249 @@
+#include "iceflow/consumer.hpp"
+#include "iceflow/dag-parser.hpp"
+#include "iceflow/iceflow.hpp"
+#include "iceflow/measurements.hpp"
+#include "iceflow/producer.hpp"
+#include "iceflow/serde.hpp"
+
+#include <ndn-cxx/face.hpp>
+#include <ndn-cxx/util/logger.hpp>
+
+#include <csignal>
+#include <functional>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/opencv.hpp>
+
+NDN_LOG_INIT(iceflow.examples.visionprocessing.faceDetection);
+
+iceflow::Measurement *measurementHandler;
+
+void signalCallbackHandler(int signum) {
+  measurementHandler->recordToFile();
+  exit(signum);
+}
+
+class FaceDetection {
+public:
+  FaceDetection(const std::string &protobufFile, const std::string &mlModel)
+      : faceNet(cv::dnn::readNet(mlModel, protobufFile)) {}
+
+  void faceDetection(std::vector<uint8_t> encodedCropped,
+                     std::function<void(std::vector<uint8_t>)> push) {
+
+    nlohmann::json deserializedData = Serde::deserialize(encodedCropped);
+
+    if (deserializedData["image"].empty()) {
+      std::cerr << "Received empty Image, cannot decode." << std::endl;
+      return;
+    } else {
+      int frameID = deserializedData["frameID"];
+      std::vector<uint8_t> encodedImage =
+          deserializedData["image"].get_binary();
+
+      // Decode the image (JPEG format) using OpenCV
```

Comment:
As above, comments like these indicate that you should consider moving this to its own method/function.